server/src/main/java/org/hawkular/HandlersManager.java

```diff
@@ -0,0 +1,136 @@
+                        return;
+                    }
+                    String baseUrl = app.baseUrl();
+                    router.route(baseUrl + "*").handler(BodyHandler.create());
+                    RestHandler handler = endpoint.getValue().newInstance();
+                    handler.initRoutes(baseUrl, router);
+                    endpoints.put(endpoint.getKey(), handler);
+                } catch (Exception e) {
+                    log.errorf(e, "Error loading Handler [%s]", endpoint);
+                }
+            });
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    public void handle(HttpServerRequest req) {
+        log.debugf("%s %s %s", req.method().name(), req.path(), req.params());
+        router.accept(req);
+    }
+
+    public void stop() {
+        applications.entrySet().stream().forEach(application -> application.getValue().stop());
+    }
+
+    @SuppressWarnings("unchecked")
+    private void scan() throws IOException {
+        String[] classpath = System.getProperty("java.class.path").split(":");
+        for (int i=0; i<classpath.length; i++) {
+            if (classpath[i].contains("hawkular") && classpath[i].endsWith("jar")) {
+                ZipInputStream zip = new ZipInputStream(new FileInputStream(classpath[i]));
+                for (ZipEntry entry = zip.getNextEntry(); entry != null; entry = zip.getNextEntry()) {
+                    if (!entry.isDirectory() && entry.getName().endsWith(".class")) {
+                        String className = entry.getName().replace('/', '.'); // including ".class"
+                        className = className.substring(0, className.length() - 6);
+                        try {
+                            Class clazz = cl.loadClass(className);
+                            Class<?>[] interfaces = clazz.getInterfaces();
+                            for (int j = 0; j < interfaces.length; j++) {
+                                if (interfaces[j].equals(BaseApplication.class)) {
+                                    String appName = clazz.getPackage().getName();
+                                    applicationsClasses.put(appName, clazz);
+                                }
+                            }
+                            if (clazz.isAnnotationPresent(RestEndpoint.class)) {
+                                RestEndpoint endpoint = (RestEndpoint)clazz.getAnnotation(RestEndpoint.class);
+                                for (int j=0; j<interfaces.length; j++) {
+                                    if (interfaces[j].equals(RestHandler.class)) {
+                                        endpointsClasses.put(endpoint.path(), clazz);
+                                    }
+                                }
+                            }
+                        } catch (Exception e) {
+                            log.errorf(e, "Error loading Handler [%s].", className);
+                            System.exit(1);
```

Comment:
At this point yes. As it sounds a fatal error (i.e. corrupted file or similar).
Another thing is if a handler has some runtime error which is handled in the start() and error in logged but no hard stop.
(Good thing of this approach is this logic can be revisit if we thing it should be different).